tsweb/precompress.go

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package tsweb
+
+import (
+	"bytes"
+	"compress/gzip"
+	"io"
+	"io/fs"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/andybalholm/brotli"
+	"golang.org/x/sync/errgroup"
+)
+
+// PrecompressDir compresses static assets in dirPath using Gzip and Brotli, so
+// that they can be later served with OpenPrecompressedFile. Meant to be run
+// during build steps, to avoid the cost of repeatedly compressing unchanging
+// resources.
+func PrecompressDir(dirPath string, options PrecompressOptions) error {
```

Comment:
kinda weird to have this in the tsweb package.

It also means all our tsweb code now brings in brotli?

Reply:
I can move it to `util/precompress`?

Reply:
Moved to `util/precompress` as discussed.